metric_tank/handler.go

```diff
@@ -46,6 +46,9 @@ func (h *Handler) HandleMessage(m *nsq.Message) error {
 	metricsReceived.Inc(int64(len(h.tmp.Metrics)))
 
 	for _, metric := range h.tmp.Metrics {
+		if metric == nil {
+			continue
+		}
 		if metric.Id == "" {
 			log.Fatal(3, "empty metric.Id - fix your datastream")
```

Comment:
I'll go through metric_tank looking for cases like this that we should be handling and recovering from and update the PR accordingly tomorrow.